app/models/plate_appearance.rb

```diff
@@ -16,6 +16,199 @@ class PlateAppearance < ActiveRecord::Base
+          end
+          if inning.man_on_third
+            inning.man_on_third = inning.man_on_first
+            inning.man_on_first = nil
+          end
+          inning.man_on_first = batter
+        elsif inning.man_on_second
+          inning.runs += 1
+          inning.man_on_second = nil
+          if inning.man_on_first
+            inning.man_on_third = inning.man_on_first
+            inning.man_on_first = nil
+          end
+          inning.man_on_first = batter
+        elsif inning.man_on_first
+          inning.man_on_third = inning.man_on_first
+          inning.man_on_first = nil
+          inning.man_on_first = batter
+        else
+          inning.man_on_first = batter
+        end
+      elsif code =~ /^SI/i || code =~ /^HBP/ || code =~ /^WALK/
+        if inning.man_on_third
+          inning.runs += 1
+          inning.man_on_third = nil
+          if inning.man_on_second
+            inning.man_on_third = inning.man_on_second
+            inning.man_on_second = nil
+          end 
+          if inning.man_on_first
+            inning.man_on_second = inning.man_on_first
+            inning.man_on_first = nil
+          end
+          inning.man_on_first = batter
+        elsif inning.man_on_second
+          inning.man_on_third = inning.man_on_second
+          inning.man_on_second = nil
+          if inning.man_on_first
+            inning.man_on_second = inning.man_on_first
+            inning.man_on_first = nil
+          end
+          inning.man_on_first = batter
+        elsif inning.man_on_first
+          inning.man_on_second = inning.man_on_first
+          inning.man_on_first = nil
+          inning.man_on_first = batter
+        else
+          inning.man_on_first = batter
+        end
+      elsif code =~ /^DO/i
+        if inning.man_on_third
+          inning.runs += 1
+          inning.man_on_third = nil
+          if inning.man_on_second
+            inning.runs += 1
+            inning.man_on_second = nil
+          end
+          if inning.man_on_first
+            inning.man_on_third = inning.man_on_first
+            inning.man_on_first = nil
+          end
+          inning.man_on_second = batter
+        elsif inning.man_on_second
+          inning.runs += 1
+          inning.man_on_second = nil
+          if inning.man_on_first
+            inning.man_on_third = inning.man_on_first
+            inning.man_on_first = nil
+          end
+          inning.man_on_second = batter
+        elsif inning.man_on_first
+          inning.man_on_third = inning.man_on_first
+          inning.man_on_first = nil
+          inning.man_on_second = batter
+        else
+          inning.man_on_second = batter
+        end
+      elsif code =~ /^TR/i
+        inning.runs += [inning.man_on_third, inning.man_on_second, inning.man_on_first].compact.size
+        inning.clear_bases
+        inning.man_on_third = batter
+      else
+        raise "You need to code: #{code}"
+      end
+    #TODO: Spacing problem
```

Comment:
Uggh, let's refactor first since I can't spot this quickly and its going away anyway...